platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/recipe/PermissionsChecker.java

```diff
@@ -10,31 +10,41 @@
  *******************************************************************************/
 package org.eclipse.che.api.machine.server.recipe;
 
+import org.eclipse.che.api.core.ForbiddenException;
 import org.eclipse.che.api.core.ServerException;
-import org.eclipse.che.api.machine.shared.ManagedRecipe;
+import org.eclipse.che.api.machine.shared.EntityPermission;
+import org.eclipse.che.api.machine.shared.dto.recipe.PermissionsDescriptor;
 
 //TODO add methods for 'all' 'any' strategies
 
 /**
- * Helps to check access to {@link ManagedRecipe}.
+ * Helps to check access to Entity.
  *
  * @author Eugene Voevodin
+ * @author Alexander Andrienko
  */
 public interface PermissionsChecker {
 
     /**
-     * Checks that user with id {@code userId} has access to {@code recipe} with given {@code permission}
+     * Checks that user with id {@code userId} has access to entity with given {@code permission}
      *
-     * @param recipe
-     *         recipe to check permissions
+     * @param target
+     *         entity to check permissions
      * @param userId
      *         user identifier
      * @param permission
-     *         permission which user should have to access recipe
+     *         permission which user should have to access entity
      * @return {@code true} when user with identifier {@code userId} has access
      * to {@code recipe} with given {@code permission} otherwise returns {@code false}
      * @throws ServerException
      *         when any error occurs while checking access
      */
-    boolean hasAccess(ManagedRecipe recipe, String userId, String permission) throws ServerException;
+    boolean hasAccess(EntityPermission target, String userId, String permission) throws ServerException;
+
+    /**
+     * User who is neither 'workspace/admin' no 'workspace/developer' can't create or
+     * update entity permissions to 'public: search', this operation allowed only
+     * for 'system/admin' or 'system/manager'
+     */
```

Comment:
I don't understand the sense of this method, this javadoc says nothing about it.
This method looks useless for me, consider removing it or making it default one and explain the contract in the better way, when `ForbiddenException` will be thrown? what is `permissions` and so on.